### PR TITLE
3.10适配鸿蒙平台

### PR DESCRIPTION
**Pull request by IceBBBBB**
3.10适配鸿蒙平台

### CODE REVIEW

File: cmake/Modules/CocosBuildHelpers.cmake [hunk 1]
```diff
@@ -139,6 +139,8 @@ elseif (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
     set(APPLE TRUE)
     set(SYSTEM_STRING "Mac OSX")
   endif()
+elseif(OHOS)
+  set(SYSTEM_STRING "OpenHarmony")
```
Comment (minggo) [inline cmake/Modules/CocosBuildHelpers.cmake:143]: Should change to HarmonyOS Next?

File: cocos/base/CCRefPtr.h [hunk 2]
```diff
@@ -211,36 +211,43 @@ template <typename T> class RefPtr
     inline bool operator > (const T * other) const { return _ptr > other; }
     
     inline bool operator > (typename std::remove_const<T>::type * other) const { return _ptr > other; }
-    
-    inline bool operator > (const std::nullptr_t other) const { return _ptr > other; }
-    
+    // TODO
+	  #if(CC_TARGET_PLATFORM != CC_PLATFORM_OHOS)
```
Comment (minggo) [inline cocos/base/CCRefPtr.h]: Error indent. Please check other codes too.

File: cocos/network/WebSocket.cpp [hunk 3]
```diff
@@ -101,15 +101,19 @@ class WebSocketCallbackWrapper {
...
                                 void *user, void *in, size_t len)
     {
         // Gets the user data from context. We know that it's a 'WebSocket' instance.
-        WebSocket* wsInstance = (WebSocket*)libwebsocket_context_user(ctx);
-        if (wsInstance)
-        {
+		#if(CC_TARGET_PLATFORM != CC_PLATFORM_OHOS)
+       		WebSocket* wsInstance = (WebSocket*)libwebsocket_context_user(ctx);
+        	if (wsInstance)
+         {
```
Comment (minggo) [inline cocos/network/WebSocket.cpp]: Wrong indent.

File: cocos/network/WebSocket.cpp [hunk 4]
```diff
@@ -311,27 +316,31 @@ bool WebSocket::init(const Delegate& delegate,
...
-
+#endif
     if (protocols && protocols->size() > 0)
     {
         int i = 0;
         for (std::vector<std::string>::const_iterator iter = protocols->begin(); iter != protocols->end(); ++iter, ++i)
         {
             char* name = new char[(*iter).length()+1];
             strcpy(name, (*iter).c_str());
+		#if(CC_TARGET_PLATFORM != CC_PLATFORM_OHOS)
```
Comment (minggo) [inline cocos/network/WebSocket.cpp]: Wrong indent.

File: cocos/network/WebSocket.cpp [hunk 5]
```diff
@@ -403,16 +412,18 @@ int WebSocket::onSubThreadLoop()
 {
     if (_readyState == State::CLOSED || _readyState == State::CLOSING)
     {
-        libwebsocket_context_destroy(_wsContext);
+		#if(CC_TARGET_PLATFORM != CC_PLATFORM_OHOS)
+        	libwebsocket_context_destroy(_wsContext);
+		 #endif
```
Comment (minggo) [inline cocos/network/WebSocket.cpp]: Wrong indent.

File: cocos/network/WebSocket.cpp [hunk 6]
```diff
@@ -520,7 +541,9 @@ int WebSocket::onSocketCallback(struct libwebsocket_context *ctx,
                  * start the ball rolling,
                  * LWS_CALLBACK_CLIENT_WRITEABLE will come next service
                  */
-                libwebsocket_callback_on_writable(ctx, wsi);
+			#if(CC_TARGET_PLATFORM != CC_PLATFORM_OHOS)
+               libwebsocket_callback_on_writable(ctx, wsi);
+			#endif
```
Comment (minggo) [inline cocos/network/WebSocket.cpp]: Wrong indent.

File: cocos/network/WebSocket.cpp [hunk 7]
```diff
@@ -574,16 +597,33 @@ int WebSocket::onSocketCallback(struct libwebsocket_context *ctx,
                         	if (remaining != n)
                         		writeProtocol |= LWS_WRITE_NO_FIN;
                         }
-
+	#if(CC_TARGET_PLATFORM != CC_PLATFORM_OHOS)
                         auto bytesWrite = libwebsocket_write(wsi,  &buf[LWS_SEND_BUFFER_PRE_PADDING], n, (libwebsocket_write_protocol)writeProtocol);
-
+
```
Comment (minggo) [inline cocos/network/WebSocket.cpp]: Unneeded indent.

File: cocos/network/WebSocket.cpp [hunk 8]
```diff
@@ -574,16 +597,33 @@ int WebSocket::onSocketCallback(struct libwebsocket_context *ctx,
...
                         auto bytesWrite = libwebsocket_write(wsi,  &buf[LWS_SEND_BUFFER_PRE_PADDING], n, (libwebsocket_write_protocol)writeProtocol);
-
+	
                         // Buffer overrun?
                         if (bytesWrite < 0)
                         {
                             //break;
+                       }
+					       // Do we have another fragments to send?
+					    else if (remaining != n)
```
Comment (minggo) [inline cocos/network/WebSocket.cpp]: Wrong indent.

File: cocos/network/WebSocket.cpp [hunk 9]
```diff
@@ -574,16 +597,33 @@ int WebSocket::onSocketCallback(struct libwebsocket_context *ctx,
...
+                            CC_SAFE_DELETE_ARRAY(data->bytes);
+                            CC_SAFE_DELETE(data);
+                            CC_SAFE_DELETE_ARRAY(buf);
+                            _wsHelper->_subThreadWsMessageQueue->erase(iter++);
+                            CC_SAFE_DELETE(subThreadMsg);
+                        }
+					   
+	#else
+	   					 // Do we have another fragments to send?
+						 if (remaining != n)
```
Comment (minggo) [inline cocos/network/WebSocket.cpp]: Wrong indent.

File: cocos/renderer/CCGLProgramState.cpp [hunk 10]
```diff
@@ -363,8 +363,11 @@ GLProgramState::GLProgramState()
 , _vertexAttribsFlags(0)
 , _glprogram(nullptr)
 , _nodeBinding(nullptr)
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_OHOS)
+, _backToForegroundlistener(nullptr)
```
Comment (minggo) [inline cocos/renderer/CCGLProgramState.cpp]: Where defines this member variable.

File: tests/cpp-tests/Classes/FontTest/FontTest.cpp [hunk 11]
```diff
@@ -67,12 +67,12 @@ void FontTest::showFont(const std::string& fontFile)
     removeChildByTag(kTagColor2, true);
     removeChildByTag(kTagColor3, true);
 
-    auto top = Label::createWithSystemFont(fontFile, fontFile, 24);
-    auto left = Label::createWithSystemFont("alignment left", fontFile, fontSize,
+    auto top = Label::createWithTTF(fontFile, fontFile, 24);
+    auto left = Label::createWithTTF("alignment left", fontFile, fontSize,
```
Comment (minggo) [inline tests/cpp-tests/Classes/FontTest/FontTest.cpp]: Why change to use ttf?

File: cmake/Modules/CocosBuildHelpers.cmake (hunk 1 above)
Comment (IceBBBBB) [reply cmake/Modules/CocosBuildHelpers.cmake:143]: > Should change to HarmonyOS Next?

Modified

File: cocos/base/CCRefPtr.h (hunk 2 above)
Comment (IceBBBBB) [reply cocos/base/CCRefPtr.h]: > Error indent. Please check other codes too.

Modified

File: cocos/network/WebSocket.cpp (hunk 3 above)
Comment (IceBBBBB) [reply cocos/network/WebSocket.cpp]: Modified

File: cocos/network/WebSocket.cpp (hunk 4 above)
Comment (IceBBBBB) [reply cocos/network/WebSocket.cpp]: Modified

File: cocos/network/WebSocket.cpp (hunk 5 above)
Comment (IceBBBBB) [reply cocos/network/WebSocket.cpp]: Modified

File: cocos/network/WebSocket.cpp (hunk 6 above)
Comment (IceBBBBB) [reply cocos/network/WebSocket.cpp]: Modified

File: cocos/renderer/CCGLProgramState.cpp (hunk 10 above)
Comment (IceBBBBB) [reply cocos/renderer/CCGLProgramState.cpp]: removed

File: tests/cpp-tests/Classes/FontTest/FontTest.cpp (hunk 11 above)
Comment (IceBBBBB) [reply tests/cpp-tests/Classes/FontTest/FontTest.cpp]: Modified

File: cocos/network/WebSocket.cpp (hunk 9 above)
Comment (IceBBBBB) [reply cocos/network/WebSocket.cpp]: Modified

File: cocos/network/WebSocket.cpp (hunk 8 above)
Comment (IceBBBBB) [reply cocos/network/WebSocket.cpp]: Modified

File: cocos/network/WebSocket.cpp (hunk 7 above)
Comment (IceBBBBB) [reply cocos/network/WebSocket.cpp]: Modified

File: cmake/Modules/CocosBuildHelpers.cmake (hunk 1 above)
Comment (minggo) [reply cmake/Modules/CocosBuildHelpers.cmake:143]: I don't see any change.

File: cocos/network/WebSocket.cpp (hunk 3 above)
Comment (minggo) [reply cocos/network/WebSocket.cpp]: I don't find any change.

File: cocos/network/WebSocket.cpp (hunk 9 above)
Comment (minggo) [reply cocos/network/WebSocket.cpp]: `if (remaining != n)` still has wrong indent.

File: cmake/Modules/CocosBuildHelpers.cmake (hunk 1 above)
Comment (IceBBBBB) [reply cmake/Modules/CocosBuildHelpers.cmake:143]: Sorry for misreading, this issue will be uniformly modified in subsequent versions

File: cocos/network/WebSocket.cpp (hunk 3 above)
Comment (IceBBBBB) [reply cocos/network/WebSocket.cpp]: I have adjusted the indentation and aligned it with the pre line and next lines. Do you want the macro to define without indentation?

File: cmake/Modules/CocosBuildHelpers.cmake (hunk 1 above)
Comment (minggo) [reply cmake/Modules/CocosBuildHelpers.cmake:143]: No. As you can see, after the modification, the result looks like
WeChatf233ec4926dad93cfb30b600a05f8d62

File: cocos/network/WebSocket.cpp (hunk 3 above)
Comment (minggo) [reply cocos/network/WebSocket.cpp]: No. As you can see, after the modification, new codes have different indent with existing codes:
WeChatf233ec4926dad93cfb30b600a05f8d62

You should see the result in web, not in local editor.

Comment (minggo) [reply cocos/network/WebSocket.cpp]: We usually use 4 spaces instead of a tab as tab may have different number of spaces in different editor.

Comment (IceBBBBB) [reply cocos/network/WebSocket.cpp]: Modified，Please check it

File: cocos/network/WebSocket.cpp (hunk 9 above)
Comment (IceBBBBB) [reply cocos/network/WebSocket.cpp]: Modified，Please check it

File: cmake/Modules/CocosBuildHelpers.cmake (hunk 1 above)
Comment (minggo) [reply cmake/Modules/CocosBuildHelpers.cmake:143]: Did you mean will not modify it in this branch or not in this pull request?

Comment (IceBBBBB) [reply cmake/Modules/CocosBuildHelpers.cmake:143]: It will be modified. It has been modified in every version, but it will be pushed later

Comment (IceBBBBB) [reply cmake/Modules/CocosBuildHelpers.cmake:143]: Modified，Please check it

**Pull request merged**